### ISSUE TITLE
Update admin app to use Angular latest

### ISSUE THREAD

**Issue by dianabarsan** [medic/cht-core#7369]
**Describe the issue**
The admin app uses old, unsupported libraries, that fewer and fewer people are familiar with, or _should get_  familiar with. 
It's riddled with gotchas (ex: https://stackoverflow.com/questions/45980598/ngannotate-warning-stringmap-expected-string-key) and antiquated code. It's the epitome of technical debt. 

**Describe the improvement you'd like**
If we plan on maintaining the admin app, we should bring it into the present and update it to use latest Angular.

**Describe alternatives you've considered**
I think it would be silly to use two different frameworks in our app, so Angular is definitely the way to go.

**Comment by dianabarsan** [comment]
The idea is to reuse as much `webapp` code as possible - we used to have resource sharing in the AngularJS implementation of both apps, and reuse as much of the setup / testing boilerplate as possible. Ideally we would not have any code duplication between webapp and admin.

**Comment by jkuester** [comment]
Do we have to keep this as a separate "app" at all?  My preference would be to just move the admin functionality into webapp.  We can put all the admin-specific functionality in a separate feature module (within webapp) that is [lazy-loaded](https://angular.dev/reference/migrations/route-lazy-loading) only for admin users.  So, we should have maximum convenience/maintainability for reusing existing webapp functionality without impacting the bundle size of the app for normal users.

I _think_ this would essentially leverage the functionality Gareth was tinkering with in https://github.com/medic/cht-core/issues/8865 (but instead of breaking up the existing webapp bundle into multiple modules, we are just keeping a new module separate...).

**Comment by jkuester** [comment]
> move the admin functionality

Also, IMHO we should be very specific about _which_ features we invest the time to move. We do not need them all.  A lot of the admin app functionality is now duplicated/replaced by cht-conf actions.

**Comment by dianabarsan** [comment]
> My preference would be to just move the admin functionality into webapp.

This is how it was before, and I agree that there are some clear advantages to having admin be a subcomponent in webapp. 
I'm not sure if it's worth the architecture shuffle in this first rewrite. I think it's sufficient that we reuse as much as possible and keep the apps as aligned as possible. I agree that further convergence is only guaranteed if they share "Angular". 

My thinking is the task is less complex if we just do a 1-1 replica, with reuse of webapp services. We can tinker with lazy loading in a 2nd iteration.

**Comment by jkuester** [comment]
I am fine with whatever way you determine will be most efficient in terms of development/maintenance effort. 👍   

Personally, I am worried you might end up expending as much effort getting the Angular build for the admin app working like you want as it would take to just do the lazy-loading in webapp (and then we still would have to go back and do the lazy-loading in the 2nd iteration anyway...). But maybe I am overthinking this.  I am still traumatized from the complexity involved in building the cht-form component... 😅

**Comment by dianabarsan** [comment]
> I am still traumatized from the complexity involved in building the cht-form component... 😅

Is the worst case scenario that we import ALL of webapp's services to admin app? :D

**Comment by dianabarsan** [comment]
I have updated the title. We are not going to Angular latest (21) because it bumps the minimum browser version to 111. 
To use Angular 21 for admin we would need to: 
1. have a different minimum browser version for Admin
this will mean that users that are on our minimum browser, and _can_ login in Webapp, won't be able to use the admin app even though they have online user accounts. 
2. the codebases of Admin and Webapp will further diverge, because we cannot bump the minimum browser for Webapp until 6.0 (which is not even planned yet). 

Besides using the latest Angular sugar (I don't mean to diminish Angular advancements, but neither do I think that 21 is twice as better than 20), I do not see any clear advantage to use Angular 21 in the Admin app versus Angular 20, especially given the downsides.

**Comment by dianabarsan** [comment]
I have created a [branch](https://github.com/medic/cht-core/tree/7369-admin-angular-upgrade) that will host all the Admin uplift code. 
the plan is to create PRs and merge them into this branch. Each PR would be rigorously reviewed, and all tests must pass before it gets merged into the [branch](https://github.com/medic/cht-core/tree/7369-admin-angular-upgrade). I am also adding branch protection rules so we don't accidentally push or merge unwanted commits. 
When work is done, we can merge this branch into `master` through a mega PR. The PR will be merged with a squash, and all authors of will be credited accordingly.

**Comment by dianabarsan** [comment]
In terms of project structure, we will try to stay as close to Webapp as possible, having common dependencies, code and testing patterns.

In the next few days I will start breaking down the work into pieces / pages and add the corresponding issues.

**Comment by andrablaj** [comment]
@dianabarsan, should this be assigned to Javier? Am I correct that this is the issue his team is working on?

**Comment by dianabarsan** [comment]
yes, you are correct. I can't assign @javier-perez unless he is subscribed to the issue.
